web-common-marlowe/src/Marlowe/Market/Contract1.purs

```diff
@@ -5,136 +5,93 @@ module Marlowe.Market.Contract1
-      fromFoldable
-        [ "Arbiter" /\ "The party that will choose who gets the money in the event of a disagreement between the \"Buyer\" and the \"Seller\" about the outcome."
-        , "Buyer" /\ "The party that wants to buy the item. Payment is made to the seller if they acknowledge receiving the item. "
-        , "Seller" /\ "The party that wants to sell the item. They receive the payment if the exchange is uneventful."
-        ]
-  , slotParameterDescriptions:
-      fromFoldable
-        [ "Buyer's deposit timeout" /\ "Deadline by which the \"Buyer\" must deposit the selling  \"Price\" in the contract."
-        , "Buyer's dispute timeout" /\ "Deadline by which, if the \"Buyer\" has not opened a dispute, the \"Seller\" will be paid."
-        , "Seller's response timeout" /\ "Deadline by which, if the \"Seller\" has not responded to the dispute, the \"Buyer\" will be refunded."
-        , "Timeout for arbitrage" /\ "Deadline by which, if the \"Arbiter\" has not resolved the dispute, the \"Buyer\" will be refunded."
-        ]
-  , valueParameterDescriptions:
-      fromFoldable
-        [ "Price" /\ "Amount of Lovelace to be paid by the \"Buyer\" for the item." ]
-  , choiceDescriptions:
-      fromFoldable
-        [ "Confirm problem" /\ "Acknowledge there was a problem and a refund must be granted."
-        , "Dismiss claim" /\ "The \"Arbiter\" does not see any problem with the exchange and the \"Seller\" must be paid."
-        , "Dispute problem" /\ "The \"Seller\" disagrees with the \"Buyer\" about the claim that something went wrong."
-        , "Everything is alright" /\ "The transaction was uneventful, \"Buyer\" agrees to pay the \"Seller\"."
-        , "Report problem" /\ "The \"Buyer\" claims not having received the product that was paid for as agreed and would like a refund."
-        ]
-  }
+metaData = Metadata.escrow
+
+ada :: Token
+ada = Token "" ""
+
+buyer :: Party
+buyer = Role "Buyer"
+
+seller :: Party
+seller = Role "Seller"
+
+arbiter :: Party
+arbiter = Role "Arbiter"
+
+price :: Value
+price = ConstantParam "Price"
+
+depositTimeout :: Timeout
+depositTimeout = SlotParam "Buyer's deposit timeout"
+
+disputeTimeout :: Timeout
+disputeTimeout = SlotParam "Buyer's dispute timeout"
+
+answerTimeout :: Timeout
+answerTimeout = SlotParam "Seller's response timeout"
+
+arbitrageTimeout :: Timeout
+arbitrageTimeout = SlotParam "Timeout for arbitrage"
+
+choice :: ChoiceName -> Party -> BigInteger -> Contract -> Case
+choice choiceName chooser choiceValue continuation =
+  Case
+    ( Choice (ChoiceId choiceName chooser)
+        [ Bound choiceValue choiceValue ]
+    )
+    continuation
+
+deposit :: Timeout -> Contract -> Contract -> Contract
+deposit timeout timeoutContinuation continuation =
+  When [ Case (Deposit seller buyer ada price) continuation ]
+    timeout
+    timeoutContinuation
+
+choices :: Timeout -> Party -> Contract -> Array (BigInteger /\ ChoiceName /\ Contract) -> Contract
```

Comment:
It's nicer to see these examples written in a higher order language such as purescript, they are more readable. But in the future, when we start developing the marketplace and these become IOHK "official" example contracts, we should see if we either add PS as a language that can compile to Marlowe or to express these contracts as haskell. For now is just something to keep in mind.